### PR TITLE
To-do-list: testing-part-2

### PR DESCRIPTION
**Pull request by sam2if**
In this project
we have added:
a function for editing the task description.
a function for updating an item's 'completed' status.
the "Clear all completed" function

### CODE REVIEW

**Review by Kossi-stack — COMMENTED** [review]
Hi Team,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
You have submitted a pair programming project without commits from other member(s) of the team. Kindly ensure that all members of the team participate in the project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## 

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @team, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Your submission has not met the project requirements, please kindly look at the screenshot below and complete the requirements.
image

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/addRemove.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    addTodo('Test task 2');
+    expect(container.innerHTML).toContain('Test task 2');
+  });
+
+  test('should add a third task to the container', () => {
+    addTodo('Test task 3');
+    const container = document.querySelector('.todoContainer');
+    expect(container.children.length).toBe(3);
+  });
+});
```
Comment (Shisui6) [inline src/addRemove.test.js]: - As the previous reviewer suggested, you still have not met the project requirements. You are supposed to add a test file for editing a task, editing a task completed status and the clear all completed button. None of these tests are present in your project. Please add them🙏
- Also, one of your test suites is failing as seen in the screenshot below. Please make sure that all tests are passing
image

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @sam2if and @HenaryKhuraijam, 

# Requires Changes 🍀
Great work on making the changes requested by a previous reviewer 👍
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet🪲.


 ## Highlights
 - Descriptive PR title and summary ✅
 - Follows Github flow ✅
 - Good commit messages ✅

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/addTask.test.js
Comment (joskalenda) [inline src/tests/addTask.test.js:1]: ❤️

Comment (sam2if) [reply src/tests/addTask.test.js:1]: You did not sat anything what is the problem?

File: src/addRemove.test.js
Comment (joskalenda) [inline src/addRemove.test.js:1]: - [ ] Sorry but this file is still causing a test case to fail, please fix this and resubmit your work

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @HenaryKhuraijam , |
|Great work on making the changes requested by a previous reviewer 👏, You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet. |
|  |

## Highlights

- [ ].. ✅ Green linters
- [ ].. ✅  Good testing using jest

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @HenaryKhuraijam , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- All tests passing ✔️ 
- No linter errors ✔️ 
- Professional README ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — COMMENTED** [review]
## Hello @HenaryKhuraijam @sam2if    👋,

I am Nemwel your reviewer.


## _This is an additional review as a result of the second opinion request._

I have claimed your second opinion request with a complaint about @Shisui6 your reviewer and inspected the project. 

**Kindly note the following:** when you request for a second opinion, you are not supposed to make any further commits to your project, close/ merge the PR, or request for another review in your dashboard. In your case, you already requested for another review after the CR (@AnselemOdims ) made the review and I am the result of the second opinion will be different from what it was supposed to be as you already "Interfered" with the state of the code by making commits and requesting for another review in your dashboard.

**What does this mean?** You will both have to work on the requested change by the last reviewer @joskalenda . Upon inspection of your project, I also got the same error that he was getting and the tests that reside in the addRemove.test.js file are not running as expected. Below is the screenshot of the error I am getting:

image

**Probable solution/ work fix for this but not a guarantee of 100% working**

This error typically occurs when running tests that involve browser-specific APIs or global objects (such as document and window) in a Node.js environment, which does not have these objects by default.

To resolve this issue, you can change the test environment to a browser-like environment such as jsdom, which is a JavaScript implementation of the DOM that can be run in Node.js. Here's how to configure Jest to use the jsdom test environment:

1. Install jsdom as a dev dependency by running npm install --save-dev jsdom.
2. In your package.json, add the following line under the testEnvironment property: ```"testEnvironment": "jsdom"```
  
image

3. Try running the tests again.

In case anything is not 100% clear tag me @Nemwel-Boniface so that I get the notification.

Keep rocking 🚀!

**Review by Nemwel-Boniface — COMMENTED** [review]
Good job guys. I can see it was approved :+1: 

Keep rocking :rocket:

File: src/addRemove.test.js
Comment (HenaryKhuraijam) [reply src/addRemove.test.js:1]: Thank you much for your support :smiley:

**Pull request merged**